src/test/java/com/kodilla/ecommercee/domain/ProductRepositoryTestSuite.java

```diff
@@ -0,0 +1,83 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+
+@SpringBootTest
+@ExtendWith(SpringExtension.class)
+class ProductRepositoryTestSuite {
+    @Autowired
+    ProductRepository productRepository;
+
+    @Test
+    void testSaveRetrieveProduct() {
+        //Given
+        Product product = new Product(null, "test name", "test description", new BigDecimal(100));
+
+        //When
+        productRepository.save(product);
+        Long id = product.getId();
+        Optional<Product> retrievedTestProduct = productRepository.findById(id);
+
+        //Then
+        Assertions.assertTrue(retrievedTestProduct.isPresent());
+        Assertions.assertEquals(product.getDescription(), retrievedTestProduct.get().getDescription());
+
+        //CleanUp
+        productRepository.deleteById(id);
+    }
+    @Test
+    void testDeleteProduct() {
+        //Given
+        Product product = new Product(null, "test name to delete", "test description to delete", new BigDecimal(200));
+        productRepository.save(product);
+        Long id = product.getId();
+
+        //When
+        productRepository.deleteById(id);
+        Optional<Product> retrievedTestProduct = productRepository.findById(id);
+
+        //Then
+        Assertions.assertFalse(retrievedTestProduct.isPresent());
+
+        //CleanUp
+        try {
```

Comment:
One time you do clean up with try catch and others without

Reply:
Its only because I test here the delete method, and in default the entity should not exist. Without "try-catch", this test unit ends with an exception.